File: src/backend/java/com/eventmaster/entities/Event.java
package com.eventmaster.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "events")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private String location;

    @Column(nullable = false)
    private LocalDateTime date;

    private LocalDateTime endDate;

    @Column(nullable = false)
    private String imageUrl;

    @ManyToOne
    @JoinColumn(name = "organizer_id", nullable = false)
    private User organizer;

    @Column(nullable = false)
    private Double ticketPrice;

    private Boolean featured;

    @Column(nullable = false)
    private String category;

    @OneToMany(mappedBy = "event", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<TicketType> ticketTypes = new ArrayList<>();

    @OneToMany(mappedBy = "event", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ScheduleItem> scheduleItems = new ArrayList<>();

    @OneToMany(mappedBy = "event")
    @JsonIgnore
    private List<Ticket> tickets = new ArrayList<>();

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (featured == null) {
            featured = false;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: src/types/Event.ts
export interface Event {
  id: number;
  title: string;
  description: string;
  location: string;
  date: string;
  endDate?: string;
  imageUrl: string;
  organizer: {
    id: number;
    name: string;
    email: string;
  };
  ticketPrice: number;
  featured?: boolean;
  category: string;
  tickets?: {
    id: number;
    name: string;
    price: number;
    quantity: number;
    description?: string;
  }[];
  schedule?: {
    id: number;
    title: string;
    startTime: string;
    endTime: string;
    description?: string;
    speaker?: string;
    location?: string;
  }[];
}

export interface Ticket {
  id: number;
  eventId: number;
  userId: number;
  ticketType: string;
  price: number;
  purchaseDate: string;
  status: 'CONFIRMED' | 'PENDING' | 'CANCELLED';
  ticketCode: string;
}